onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -856,6 +866,15 @@ def _infer_Pool(self, node):
                 helper.make_tensor_value_info(o, vi.type.tensor_type.elem_type,
                                               get_shape_from_sympy_shape(sympy_shape)))
 
+    def _infer_BatchNormalization(self, node):
+        self._propagate_shape_and_type(node)
+
+        # this works for opsets < 14 and 14 since we check i < len(node.output) in the loop
+        for i in [1, 2, 3, 4]:
+            if i < len(node.output):
```

Comment:
>if i < len(node.output): [](start = 12, length = 24)

This may have problem since node outputs could have empty string in the middle to omit certain optional outputs. I think the propagation should check node.output[i] not being empty instead.